resolver/dns/go19up.go

```diff
@@ -0,0 +1,44 @@
+// +build go1.9
+
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package dns
+
+import (
+	"net"
+	"strings"
+
+	"golang.org/x/net/context"
+)
+
+func customAuthorityResolver(authority string) (netResolver, error) {
+
+	if _, _, err := net.SplitHostPort(authority); err != nil && strings.Contains(err.Error(), "missing port in address") {
```

Comment:
It's better not comparing the error string directly, as the library may change the error string in the future and break this. You can do something similar to this:
https://github.com/grpc/grpc-go/blob/9907ae0e57ce159e835381ed54eb271206bf7b12/resolver/dns/dns_resolver.go#L321-L336
And in case the authority is a malformed IP address, return an error. We may want to fallback to default dns resolver if the authority is invalid.